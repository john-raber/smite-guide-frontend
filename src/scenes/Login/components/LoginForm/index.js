import React, { Component } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { Form, Button } from "reactstrap";

import EmailInput from "../EmailInput/index";
import PasswordInput from "../PasswordInput/index";

import { loginUser } from "../../services/currentUser/actions/index";

class LoginForm extends Component {
  state = {
    email: "",
    password: ""
  };

  handleFormSubmit = e => {
    const { loginUser, history } = this.props;

    e.preventDefault();

    loginUser({ user: this.state }).then(history.push("/profile"));
  };

  handleInputChange = (name, value) => {
    this.setState({
      [name]: value
    });
  };

  render() {
    return (
      <Form className="LoginForm" onSubmit={this.handleFormSubmit}>
        <EmailInput
          value={this.state.email}
          onChange={this.handleInputChange}
        />
        <PasswordInput
          value={this.state.password}
          onChange={this.handleInputChange}
        />
        <Button color="warning">Login</Button>
      </Form>
    );
  }
}

export default withRouter(
  connect(
    null,
    { loginUser }
  )(LoginForm)
);

export { LoginForm };
