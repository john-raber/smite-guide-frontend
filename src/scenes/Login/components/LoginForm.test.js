import React from "react";
import { shallow } from "enzyme";

import { LoginForm } from "./LoginForm/index";

describe("LoginForm", () => {
  it("renders correctly", () => {
    let wrapper = shallow(<LoginForm />);

    expect(wrapper.find(".LoginForm").length).toEqual(1);
  });

  it("matches its snapshot", () => {
    let wrapper = shallow(<LoginForm />);

    expect(wrapper).toMatchSnapshot();
  });
});
