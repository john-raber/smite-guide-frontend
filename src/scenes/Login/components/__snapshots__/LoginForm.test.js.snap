// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm matches its snapshot 1`] = `
<Form
  className="LoginForm"
  onSubmit={[Function]}
  tag="form"
>
  <EmailInput
    onChange={[Function]}
    value=""
  />
  <PasswordInput
    onChange={[Function]}
    value=""
  />
  <Button
    color="warning"
    tag="button"
  >
    Login
  </Button>
</Form>
`;
